bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/CreateAdvBuilder.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Builder-style interface to create new ledgers.
+ *
+ * @since 4.6
+ */
+public interface CreateAdvBuilder extends OpBuilder<WriteAdvHandle> {
```

Comment:
shouldn't it extend CreateBuilder?

Reply:
WriteAdvHandle and WriteHandle have different semantics and so we are creating separate APIs.
Therefore the CompletableFuture<T> execute method for CreateAdvBuilder is CompletableFuture<WriteAdvHandle> execute() and for CreateBuilder is CompletableFure<WriteHandle> so signatures are not compatible